src/utilities/hooks.ts

```diff
@@ -43,7 +43,7 @@ export async function runBeforeRouteHooks({ to, from, hooks }: BeforeContext): P
     if (error instanceof RouterPushError) {
       return {
         status: 'PUSH',
-        to: error.to,
+        to: error.to as Parameters<RegisteredRouterPush>,
```

Comment:
ah yep you ran into the same issue here that I ran into in https://github.com/kitbagjs/router/pull/117